charms/layers/jupyter-notebook/templates/upstart.conf

```diff
@@ -0,0 +1,11 @@
+#!upstart
+description "jupyter"
+
+start on startup
+stop on shutdown
+
+respawn
+
+script
+jupyter notebook --no-browser
```

Comment:
Ok, the charm hooks always execute as root, and you start jupyter as root, so it's best to use `/root` as home and remove the config option.